language/borrow-graph/src/graph.rs

```diff
@@ -294,9 +310,46 @@ impl<Loc: Copy, Lbl: Clone + Ord> BorrowGraph<Loc, Lbl> {
                 }
             }
         }
+        debug_assert!(joined.check_invariant());
         joined
     }
 
+    //**********************************************************************************************
+    // Consistency/Invariants
+    //**********************************************************************************************
+
+    fn check_invariant(&self) -> bool {
```

Comment:
I wonder if there is an easy declarative way to check acyclicity of the borrow graph.

Reply:
I don't think so, without running full cycle detection... For another time